employee-rostering-gwtui/src/main/java/org/optaweb/employeerostering/gwtui/client/viewport/shiftroster/ShiftGridObject.java

```diff
@@ -99,87 +108,143 @@ private void refresh() {
-                                                                                           " does not match Rotation Employee (" +
-                                                                                           penalty.getShift().getRotationEmployee().getName() + ") (Score: " + penalty.getScore() + ").");
+                                " does not match Rotation Employee (" +
+                                penalty.getShift().getRotationEmployee().getName() + ") (Score: " + penalty.getScore().toShortString() + ").");
                     } else {
                         indictmentIcons.add(IconContainer.Icon.ROTATION_VIOLATION_PENALTY, "Shift is unassigned although it has a Rotation Employee (" +
-                                                                                           penalty.getShift().getRotationEmployee().getName() + ") (Score: " + penalty.getScore() + ").");
+                                penalty.getShift().getRotationEmployee().getName() + ") (Score: " + penalty.getScore().toShortString() + ").");
                     }
                 }
-                
+
                 for (UnassignedShiftPenalty penalty : shiftView.getUnassignedShiftPenaltyList()) {
-                    indictmentIcons.add(IconContainer.Icon.UNASSIGNED_SHIFT_PENALTY, "Shift is unassigned (Score: " + penalty.getScore() + ").");
+                    indictmentIcons.add(IconContainer.Icon.UNASSIGNED_SHIFT_PENALTY, "Shift is unassigned (Score: " + penalty.getScore().toShortString() + ").");
                 }
             }
         }
     }
 
+    private String getNiceString(Shift shift) {
+        return shift.getSpot().getName() + ":" + getNiceString(shift.getStartDateTime()) + " to " + getNiceString(shift.getEndDateTime());
+    }
+
+    private String getNiceString(OffsetDateTime dateTime) {
+        DateTimeFormat timeFormat = DateTimeFormat.getFormat(PredefinedFormat.DATE_TIME_SHORT);
+        return timeFormat.format(GwtJavaTimeWorkaroundUtil.toDate(GwtJavaTimeWorkaroundUtil.toLocalDateTime(dateTime)));
+    }
+
+    private void updateColor(HardMediumSoftLongScore score) {
+        if (score.getHardScore() < 0) {
+            setAnimationDelay(mapToRange(-score.getHardScore(), 5, 0, 10));
+        } else if (score.getHardScore() > 0) {
+            setAnimationDelay(mapToRange(score.getHardScore(), 85, 100, 10));
+        } else if (score.getMediumScore() < 0) {
+            setAnimationDelay(mapToRange(-score.getMediumScore(), 30, 15, 1));
+        } else if (score.getMediumScore() > 0) {
+            setAnimationDelay(mapToRange(score.getMediumScore(), 70, 85, 10));
+        } else if (score.getSoftScore() < 0) {
+            setAnimationDelay(mapToRange(-score.getSoftScore(), 50, 30, 100));
+        } else if (score.getSoftScore() > 0) {
+            setAnimationDelay(mapToRange(score.getSoftScore(), 50, 70, 30));
```

Comment:
It seems weird that the animation delay differs based on the score. I would presume the color differs, but not the delay of an animation.
The function "mapToRange" is not self explanatory.

Reply:
`animation-delay` is what controls color and allows it to smoothly transition between different shades. Basically, all shifts are in a paused CSS "animation" where it changes color. Setting the `animation-delay` allows us to control where the animation is. This allow a shift with `20 soft` to be more green than a shift with `10 soft`, or a shift with `-10 hard` to be more red than a shift with `-1 hard`. `mapToRange` is essentially a `lerp` (Linear interpolation) between `start` and and `end` by `value/maxValue`.